spark/v3.3/spark-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ExtendedDistributionAndOrderingUtils.scala

```diff
@@ -57,10 +61,34 @@ object ExtendedDistributionAndOrderingUtils {
         } else {
           conf.numShufflePartitions
         }
-        // the conversion to catalyst expressions above produces SortOrder expressions
-        // for OrderedDistribution and generic expressions for ClusteredDistribution
-        // this allows RepartitionByExpression to pick either range or hash partitioning
-        RepartitionByExpression(ArraySeq.unsafeWrapArray(distribution), query, finalNumPartitions)
+
+        val tableProperties = table match {
+          case d : RowLevelOperationTable => d.table.properties()
+          case _ : Table => table.properties()
+        }
+
+        val isHashDistributionMode = write.requiredDistribution match {
+          case _ : ClusteredDistribution => true
+          case _ => false
+        }
+
+        val strictDistributionMode = tableProperties
+          .getOrDefault(TableProperties.STRICT_TABLE_DISTRIBUTION_AND_ORDERING,
+            TableProperties.STRICT_TABLE_DISTRIBUTION_AND_ORDERING_DEFAULT)
+
+        if (strictDistributionMode.equals("false") && isHashDistributionMode) {
+          // if strict distribution mode is not enabled, then we fallback to spark AQE
+          // to determine the number of partitions by colaesceing and un-skewing partitions
+          // Also to note, Rebalance is only supported for hash distribution mode till spark 3.3
```

Comment:
Till Spark 3.3? Or 3.4? I am confused cause this change is for 3.3.

Reply:
@aokolnychyi in spark 3.4 i see both hash and distribution mode supported for rebalance operator. I will change the statement to `in spark 3.3.`

Reply:
That's OK. Let's just fix the comment then cause it states that Spark 3.3 is supposed to work.